packages/eslint-plugin/utils/get-text-content-from-node.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Returns the actual text content from an argument passed to a translation function.
+ *
+ * @see eslint-plugin-wpcalypso
+ *
+ * @param  {Object} node    A Literal, TemplateLiteral or BinaryExpression (+) node
+ * @return {string|boolean} The concatenated string or false.
+ */
+function getTextContentFromNode( node ) {
+	if ( 'Literal' === node.type ) {
+		return node.value;
+	}
+
+	if ( 'BinaryExpression' === node.type && '+' === node.operator ) {
+		const left = getTextContentFromNode( node.left );
+		const right = getTextContentFromNode( node.right );
+
+		if ( left === false || right === false ) {
+			return false;
+		}
+
+		return left + right;
+	}
+
+	if ( node.type === 'TemplateLiteral' ) {
+		return node.quasis
+			.map( function( quasis ) {
+				return quasis.value.raw;
+			} )
```

Comment:
Nit: With arrow functions, I'd probably not expect much use for anonymous function expression ever again.

```suggestion
			.map( ( quasis ) => quasis.value.raw )
```